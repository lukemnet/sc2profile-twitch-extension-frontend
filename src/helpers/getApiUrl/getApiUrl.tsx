import api from 'config/api';
import {
  FetchMethod,
  ApiUrlType,
  ApiActivity,
} from 'types';

interface GetApiUrlParams {
  type: ApiUrlType;
  activity: ApiActivity;
  channelId: string;
}

const getApiUrl = ({ channelId, type, activity }: GetApiUrlParams) => {
  const {
    url,
    version,
    endpoints,
  } = api;
  const endpointByActivity = (type === 'config')
    ? endpoints.config[activity]
    : endpoints.viewer.get;
  const apiUrl = `${url}/${version}/${endpointByActivity.url}/${channelId}`;
  const method = endpointByActivity.method as FetchMethod;
  return {
    url: apiUrl,
    method,
  };
};

export default getApiUrl;
