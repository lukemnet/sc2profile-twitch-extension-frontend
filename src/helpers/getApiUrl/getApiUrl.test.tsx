import getApiUrl, { ApiUrlType, ApiActivity } from './getApiUrl';

const types = [
  'config',
  'viewer',
] as ApiUrlType[];

const activities = [
  'get',
  'save',
] as ApiActivity[];

it('returns correct API url', () => {
  types.map(type =>
    activities.map(activity =>
      expect(
        getApiUrl({
          channelId: '123',
          type,
          activity,
        })
      ).toMatchSnapshot()
    )
  );
});