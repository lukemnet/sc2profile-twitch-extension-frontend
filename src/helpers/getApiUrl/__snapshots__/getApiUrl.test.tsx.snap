// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns correct API url 1`] = `
Object {
  "method": "GET",
  "url": "https://localhost:8883/v2/config/123",
}
`;

exports[`returns correct API url 2`] = `
Object {
  "method": "POST",
  "url": "https://localhost:8883/v2/config/123",
}
`;

exports[`returns correct API url 3`] = `
Object {
  "method": "GET",
  "url": "https://localhost:8883/v2/viewer/123",
}
`;

exports[`returns correct API url 4`] = `
Object {
  "method": "GET",
  "url": "https://localhost:8883/v2/viewer/123",
}
`;

exports[`returns correct API url when no API version is provided 1`] = `
Object {
  "method": "GET",
  "url": "https://localhost:8883/v2/config/123",
}
`;

exports[`returns correct API url when no API version is provided 2`] = `
Object {
  "method": "POST",
  "url": "https://localhost:8883/v2/config/123",
}
`;

exports[`returns correct API url when no API version is provided 3`] = `
Object {
  "method": "GET",
  "url": "https://localhost:8883/v2/viewer/123",
}
`;

exports[`returns correct API url when no API version is provided 4`] = `
Object {
  "method": "GET",
  "url": "https://localhost:8883/v2/viewer/123",
}
`;
